### ISSUE TITLE
NS6 on iOS13 - hasDifferentColorAppearanceComparedToTraitCollection is not a function

### ISSUE THREAD

**Issue by dpdragnev** [NativeScript/sidekick-feedback#436]
I am using NS 6.2.0 on a Windows 10 machine, testing on a physical device with iOS 13.1.2.  When I try to build and run the app via Sidekick, I see that the app is built successfully and installed on the device.

When I start it, the app crashes immediately.  I checked the device logs and I saw the following error:

> CONSOLE LOG file:///app/bundle.js:555:16: TypeError: this.traitCollection.hasDifferentColorAppearanceComparedToTraitCollection is not a function. (In 'this.traitCollection.hasDifferentColorAppearanceComparedToTraitCollection(previousTraitCollection)', 'this.traitCollection.hasDifferentColorAppearanceComparedToTraitCollection' is undefined) -- undefined
***** Fatal JavaScript exception - application has been terminated. *****

Here is my package.json:
```
{
  "name": "app",
  "version": "4.5.0",
  "description": "",
  "license": "SEE LICENSE IN <your-license-filename>",
  "readme": "NativeScript Application",
  "repository": "<fill-your-repository-here>",
  "nativescript": {
    "id": "com.company.app",
    "tns-ios": {
      "version": "6.2.0"
    }
  },
  "dependencies": {
    "@types/node": "^12.12.7",
    "moment": "2.20.1",
    "moment-timezone": "0.5.14",
    "nativescript": "^6.2.0",
    "nativescript-appversion": "1.4.1",
    "nativescript-email": "1.5.1",
    "nativescript-exit": "1.0.1",
    "nativescript-fingerprint-auth": "6.0.3",
    "nativescript-image-zoom": "^3.0.3",
    "nativescript-iqkeyboardmanager": "1.5.1",
    "nativescript-loading-indicator": "2.4.0",
    "nativescript-masked-text-field": "4.0.3",
    "nativescript-numeric-keyboard": "4.3.1",
    "nativescript-pdf-view": "2.0.1",
    "nativescript-phone": "1.4.0",
    "nativescript-plugin-firebase": "^9.1.1",
    "nativescript-theme-core": "1.0.4",
    "nativescript-ui-calendar": "5.0.0",
    "nativescript-zendesk-sdk": "^0.2.0",
    "tns-core-modules": "6.2.1"
  },
  "devDependencies": {
    "babel-traverse": "6.4.5",
    "babel-types": "6.4.5",
    "babylon": "6.4.5",
    "lazy": "1.0.11",
    "nativescript-dev-webpack": "^1.1.1",
    "tns-platform-declarations": "6.0.1",
    "typescript": "3.4.5"
  }
}

```

If I run the app through the CLI on Android, everything runs fine.

Any ideas how to fix that?

**Comment by NickIliev** [comment]
@dpdragnev you are hitting [this issue](https://github.com/NativeScript/NativeScript/issues/8082) and the modules team introduced [this fix](https://github.com/NativeScript/NativeScript/pull/8093) which will be included in the upcoming release. Keep track on the PR - once it is merged into master it will be available with tns-core-modules@rc (and later in the upcoming release)

**Comment by dpdragnev** [comment]
Thank you @NickIliev 
Do you have timeframe for this fix to be released?  Or, perhaps a workaround?  We are supposed to release a new version in a few weeks and as things are now, I cannot do that.

**Comment by NickIliev** [comment]
@dpdragnev there are still some tests that need to pass - I would say that the fix will be released next week.

**Comment by dpdragnev** [comment]
@NickIliev I saw that the PR was merged today.  I got latest (6.2.1), but I am still getting the same error:

```
CONSOLE LOG file:///app/bundle.js:604:16: TypeError: this.traitCollection.hasDifferentColorAppearanceComparedToTraitCollection is not a function. (In 'this.traitCollection.hasDifferentColorAppearanceComparedToTraitCollection(previousTraitCollection)', 'this.traitCollection.hasDifferentColorAppearanceComparedToTraitCollection' is undefined) -- undefined
***** Fatal JavaScript exception - application has been terminated. *****
```

Do you have any suggestion on how to solve this.  I need to push a new version of our app in 2 weeks and need to get the latest code to our QA.

Please advise.

**Comment by NickIliev** [comment]
@dpdragnev the PR is merged into release and is currently located in the **rc** version (**not** in the latest officially published version). Test  with the RC as follows:

```
npm i tns-core-modules@rc --save
```

**Comment by dpdragnev** [comment]
@NickIliev No luck.  I am using:
```
"tns-core-modules": "^6.2.1-rc-2019-11-18-144148-01"
```
but I am still getting the same error:

```
[19-11-19 06:59:12.648] (CLI) Starting publishing.
[19-11-19 06:59:14.830] (CLI) Build machine host name: administrators-Mac-19.local
[19-11-19 06:59:35.389] Spawn failed.
, Error: Spawn failed.
    at CloudPublishService.getiOSError (C:\Users\daniel.dragnev\AppData\Roaming\.nativescript-cli\extensions\node_modules\nativescript-cloud\lib\services\cloud-publish-service.js:79:21)
    at CloudPublishService.<anonymous> (C:\Users\daniel.dragnev\AppData\Roaming\.nativescript-cli\extensions\node_modules\nativescript-cloud\lib\services\cloud-publish-service.js:96:29)
    at Generator.next (<anonymous>)
    at fulfilled (C:\Users\daniel.dragnev\AppData\Roaming\.nativescript-cli\extensions\node_modules\nativescript-cloud\lib\services\cloud-publish-service.js:4:58)
```

Please advise.

**Comment by NickIliev** [comment]
@dpdragnev the last error doesn't look related to `TypeError: this.traitCollection.hasDifferentColorAppearanceComparedToTraitCollection is not a function.`. 

Can you post the whole log - meanwhile verify that you are using the latest CLI (tns --version) and if needed reinstall the NativeScript CLI (npm i -g nativescript@latest) and then make a clean rebuild. Also, let me know what is the error appearing for debug build or for release (or both).

**Comment by dpdragnev** [comment]
Thank you @NickIliev I just updated the CLI and I am building now.  Will send you the complete log in a minute.  As for the build, the build (release) actually completes without an issue.  The error happens after I enter my iTunesConnect credentials so Sidekick can publish the app.

**Comment by dpdragnev** [comment]
@NickIliev The complete log is attached
[iOSPublishLog.txt](https://github.com/NativeScript/sidekick-feedback/files/3864724/iOSPublishLog.txt)

**Comment by dpdragnev** [comment]
@NickIliev as a side note:  I was able to build and push the app from my Mac using Xcode.  So, the issue is definitely somewhere in the publish process when using Sidekick.

**Comment by dpdragnev** [comment]
@NickIliev To go back to our original error.  I loaded the app on my Mac, upgraded NS to 6.2.1, upgraded the tns-core-modules to the 6.2.1-rc-2019-11-18-144148-01 version, upgraded my Xcode to 10.3, but I am still getting the same error.

I am attaching the complete log of the crash.
[iOSCrash.txt](https://github.com/NativeScript/sidekick-feedback/files/3866103/iOSCrash.txt)

As a side note I wanted to mention that I have issues building Android in release mode as well, but I will open a separate issue for that.

Please let me know how to proceed.

Thank you.

**Comment by NickIliev** [comment]
@dpdragnev I can confirm that indeed the issue is reproducible on our side and that the team is working on a fix. The issue is [related to this one](https://github.com/NativeScript/sidekick-feedback/issues/435#issuecomment-553250450). Once we have a stable solution we will publish a new version of Sidekick and I will update the information in the linked issues.

Meanwhile, as a temporary workaround use local Mac builds to publish for iOS.

**Comment by dpdragnev** [comment]
@NickIliev Thank you.  

The problem is that even building locally, the app still crashes when I run it.  See my previous message for the log of the crash.

Do you have any ideas on how to fix that?

**Comment by NickIliev** [comment]
@dpdragnev can you post the content of your `package.json`?

**Comment by dpdragnev** [comment]
@NickIliev 

Here is it:

```
{
  "name": "app",
  "version": "4.5.0",
  "description": "",
  "license": "SEE LICENSE IN <your-license-filename>",
  "readme": "NativeScript Application",
  "repository": "<fill-your-repository-here>",
  "nativescript": {
    "id": "com.company.app"
  },
  "dependencies": {
    "@types/node": "^12.12.7",
    "moment": "2.20.1",
    "moment-timezone": "0.5.14",
    "nativescript-appversion": "1.4.1",
    "nativescript-email": "1.5.1",
    "nativescript-exit": "1.0.1",
    "nativescript-fingerprint-auth": "6.0.3",
    "nativescript-image-zoom": "^3.0.3",
    "nativescript-iqkeyboardmanager": "1.5.1",
    "nativescript-loading-indicator": "2.4.0",
    "nativescript-masked-text-field": "4.0.3",
    "nativescript-numeric-keyboard": "4.3.1",
    "nativescript-pdf-view": "2.0.1",
    "nativescript-phone": "1.4.0",
    "nativescript-plugin-firebase": "^9.1.1",
    "nativescript-theme-core": "1.0.4",
    "nativescript-ui-calendar": "5.0.0",
    "nativescript-zendesk-sdk": "^0.2.0",
    "tns-core-modules": "^6.2.1-rc-2019-11-18-144148-01"
  },
  "devDependencies": {
    "babel-traverse": "6.4.5",
    "babel-types": "6.4.5",
    "babylon": "6.4.5",
    "lazy": "1.0.11",
    "nativescript-dev-webpack": "1.3.0",
    "tns-platform-declarations": "6.2.1",
    "typescript": "3.4.5"
  }
}
```

**Comment by dpdragnev** [comment]
@NickIliev after upgrading my Mac's OS (to Catalina), the version of Xcode, and all plugins, the app works again.

Thank you for your assistance

**Comment by NickIliev** [comment]
@dpdragnev glad tao hear that you were able to resolve the issue - I was thinking about what you are probably building with an older version of XCode which was brining an older API - still, the fix should have worked with an older XCode version so again thanks for reporting the issue

**Comment by stumpykilo** [comment]
I'm getting the error when using Sidekick on Windows. Do you know when Sidekick will get a new release to fix this issue?

> JavaScript error:
> file:///app/vendor.js:27839:93: JS ERROR TypeError: this.traitCollection.hasDifferentColorAppearanceComparedToTraitCollection is not a function.

I tried what @dpdragnev did in his package.json with no success. `"tns-core-modules": "^6.2.1-rc-2019-11-18-144148-01"`
